framework/Supports/Autoptimize.php

```diff
@@ -130,6 +129,50 @@ public function add_dns_prefetch( $content ) {
 		return $content;
 	}
 
+	/**
+	 * Add target and rel for links.
+	 *
+	 * @param string $content HTML content generated for the page.
+	 *
+	 * @return string
+	 */
+	public function add_target_rel( $content ) {
+		$tags_matches = [];
+
+		if ( preg_match_all( '#<a.*>.*<\/a>#Usmi', $content, $matches ) ) {
+			$tags_matches = array_merge( $tags_matches, $matches[0] );
+		}
+
+		if ( empty( $tags_matches ) ) {
+			return $content;
+		}
+
+		preg_match( '#http(s)?\:\/\/(([a-z0-9\_\-\.]+)\.([a-z0-9]{2,5}))\/?#', site_url(), $site_domain );
+
+		foreach ( $tags_matches as $tag ) {
+			if ( preg_match( '#href="(http\:\/\/|https\:\/\/|\/\/)(([a-z0-9\_\-\.]+)\.([a-z0-9]{2,5})).*"#', $tag, $domain ) ) {
+				if ( false !== strpos( $domain[2], $site_domain[2] ) ) {
+					continue;
+				}
+
+				$basic_tag = $tag;
+
+				if ( ! preg_match( '#rel="(.*)"#', $basic_tag, $rel_domain ) ) {
+					$tag = str_replace( $domain[0], $domain[0] . ' rel="noopener noreferrer"', $tag );
+				}
+
+				if ( ! preg_match( '#target="(.*)"#', $basic_tag, $target_domain ) ) {
+					$tag = str_replace( $domain[0], $domain[0] . ' target="_blank"', $tag );
+				}
```

Comment:
I think these regular expressions can work buggy because of "greedy" behaviour. Please check what you will get if you have <a target="..." rel="..." href="..."> (manually edit inside editors code)